bolt-contracts/src/contracts/BoltParameters.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+
+/// @title Bolt Parameters
+/// @notice The BoltParameters contract contains all the parameters for the Bolt protocol.
+/// @dev This contract is upgradeable using the UUPSProxy pattern. Storage layout remains fixed across upgrades
+/// with the use of storage gaps.
+/// See https://docs.openzeppelin.com/contracts/4.x/upgradeable#storage_gaps
+contract BoltParameters is OwnableUpgradeable, UUPSUpgradeable {
+    // =========== CONSTANTS ========= //
+    /// @dev See EIP-4788 for more info
+    address internal constant BEACON_ROOTS_CONTRACT = 0x000F3df6D732807Ef1319fB7B8bB8522d0Beac02;
+
+    /// @notice The EIP-4788 time window in slots
+    uint256 internal constant EIP4788_WINDOW = 8191;
+    // =========== STORAGE =========== //
+
+    // --> Storage layout marker: 0 bits
+
+    /// @notice Duration of an epoch in seconds.
+    uint48 public EPOCH_DURATION;
+
+    /// @notice Duration of the slashing window in seconds.
+    uint48 public SLASHING_WINDOW;
+
+    /// @notice Whether to allow unsafe registration of validators
+    /// @dev Until the BLS12_381 precompile is live, we need to allow unsafe registration
+    /// which means we don't check the BLS signature of the validator pubkey.
+    bool public ALLOW_UNSAFE_REGISTRATION;
+    // --> Storage layout marker: 48 + 48 + 8 = 104 bits
+
+    /// @notice The maximum duration of a challenge before it is automatically considered valid.
+    uint48 public MAX_CHALLENGE_DURATION;
+
+    /// @notice The challenge bond required to open a challenge.
+    uint256 public CHALLENGE_BOND;
+
+    /// @notice The maximum number of blocks to look back for block hashes in the EVM.
+    uint256 public BLOCKHASH_EVM_LOOKBACK;
+
+    /// @notice The number of slots to wait before considering a block justified by LMD-GHOST.
+    uint256 public JUSTIFICATION_DELAY;
+
+    /// @notice The timestamp of the eth2 genesis block.
+    uint256 public ETH2_GENESIS_TIMESTAMP;
+
+    /// @notice The duration of a slot in seconds.
+    uint256 public SLOT_TIME;
+    // --> Storage layout marker: 6 words
```

Comment:
This is good, but to make sure this is right I suggest using solc json storage layout outputs.
ref: https://docs.soliditylang.org/en/latest/using-the-compiler.html#compiler-input-and-output-json-description
and https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html#json-output

Reply:
This is sick: `forge inspect BoltParameters storage-layout --pretty`

Reply:
TIL!